integrations/langflow/tests/integration/test_langflow_integration.py

```diff
@@ -0,0 +1,352 @@
+        test_files = list(langflow_fixtures_dir.glob("*.json"))
+        
+        for test_file in test_files:
+            with open(test_file, "r") as f:
+                langflow_data = json.load(f)
+            
+            try:
+                # Analyze
+                analysis = converter.analyze(langflow_data)
+                
+                # Convert
+                workflow = converter.convert(langflow_data)
+                
+                # Check consistency
+                assert analysis["node_count"] == len(workflow.steps)
+                
+                # Total components should match steps
+                total_components = sum(analysis["component_types"].values())
+                assert total_components == len(workflow.steps)
+                
+            except (ConversionError, ValidationError):
+                # Some test files might be designed to fail
+                continue
+    
+    @pytest.mark.slow
+    def test_performance_benchmarks(
+        self, 
+        langflow_fixtures_dir: Path, 
+        converter: LangflowConverter,
+        expectations: Dict[str, Any]
+    ):
+        """Test performance benchmarks for different workflow sizes."""
+        perf_expectations = expectations.get("performance_expectations", {})
+        
+        test_files = list(langflow_fixtures_dir.glob("*.json"))
+        
+        for test_file in test_files:
+            with open(test_file, "r") as f:
+                langflow_data = json.load(f)
+            
+            # Categorize workflow size
+            node_count = len(langflow_data.get("data", {}).get("nodes", []))
+            
+            if node_count <= 3:
+                size_category = "small_workflow"
+            elif node_count <= 10:
+                size_category = "medium_workflow"
+            else:
+                size_category = "large_workflow"
+            
+            if size_category not in perf_expectations:
+                continue
+            
+            expectations_for_size = perf_expectations[size_category]
+            max_time = expectations_for_size.get("max_conversion_time", 10.0)
+            
+            # Measure conversion time
+            start_time = time.time()
+            try:
+                workflow = converter.convert(langflow_data)
+                conversion_time = time.time() - start_time
+                
+                assert conversion_time < max_time, (
+                    f"Conversion took {conversion_time:.2f}s, expected < {max_time}s "
+                    f"for {size_category} ({node_count} nodes)"
+                )
+                
+            except (ConversionError, ValidationError):
+                # Skip performance test for workflows that don't convert
+                continue
+    
+    @pytest.mark.slow
+    @pytest.mark.asyncio
+    @patch('stepflow_langflow_integration.executor.udf_executor.UDFExecutor._execute_langflow_component')
+    async def test_execution_with_mocking(
```

Comment:
Todo what is this testing? It seems to be mocking the UDF executor at the component level? How does this actually inject these mock behaviors?